.goreleaser.yml

```diff
@@ -0,0 +1,22 @@
+builds:
+  - goos:
+      - linux
+      - darwin
+    goarch:
+      - amd64
+    ldflags:
+      - -s -w -X main.version={{.Version}} -X main.build={{.ShortCommit}}
```

Comment:
As we are creating releases only on tag, I think we can skip using commit-sha in version. Agree ?

Reply:
I used Makefile reference https://github.com/gojekfarm/stevedore/blob/master/Makefile#L17 for passing version and build variables. From the version and build number it looks like we are going to follow semantic version `Major.Minor.Patch` where `Patch=commit-sha`. Is my assumption correct?

Reply:
@arunvelsriram yes your assumption is correct.

@dineshba since we will be doing a release only for tags, it should be fine, if there is a build created from a not tag, build will append commit-sha, this will be useful to identify if the builds are generated from tag or not

Reply:
@arunvelsriram this will happen if the binary is pre-compiled, with `brew` we can install after compiling from the source with that we won't face this issue. if we are going with `pre-compiled` then we have use `brew cask` and have to bypass the warning

Reply:
@arunvelsriram can use the makefile to build the image?  github actions could be setting appropriate env variable for goos and goarch, if we just use the build stevedore, we can leverage the makefile build/compile target

Reply:
I don't see a way to directly integrate it in to goreleaser.

It has way to skip building binaries but then we might have to do something to make the binaries available during publish.

Reply:
using goreleaser for build has several advantages like easy way to generate multiple platforms and architectures, archiving the binaries with extra files and cutomizable archive name for different architectures and platforms.

Reply:
@arunvelsriram lets use goreleaser for every tagged release, if we can't pass the argument I think it should be okay, we can use the build from makefile for development purpose alone.

Reply:
okay. generating dev/master builds is a nice idea. please validate my understanding:

tagged commits on master - create binaries, draft release in github using goreleaser

commits on master -  create dev/master binaries preferably using goreleaser (and make it available to users as CI artifacts)

Reply:
@arunvelsriram your understanding is correct, on every tag lets create a proper release in Github using goreleaser

for local build and artifacts via CI, we can try to use goreleaser, if not we can fall back to the current way of generating them

Reply:
@jskswamy I have pushed changes for building master binaries and publishing them as CI artifacts. I have used [goreleaser's snapshot mode](https://goreleaser.com/customization/snapshots/) to create OS/Platform specific archived binaries. Then stored the archives as artifacts in the CI.

Link to sample run: 
https://github.com/arunvelsriram/stevedore/actions/runs/349964113
https://github.com/arunvelsriram/stevedore/runs/1365060854?check_suite_focus=true

Reply:
@jskswamy @dineshba please share your thoughts

Reply:
LGTM 👍🏽

Reply:
@arunvelsriram looks good to me